src/wp-includes/option.php

```diff
@@ -1529,6 +1530,20 @@ function set_transient( $transient, $value, $expiration = 0 ) {
 	 */
 	$expiration = apply_filters( "expiration_of_transient_{$transient}", $expiration, $value, $transient );
 
+	$autoload = is_bool( $autoload ) ? $autoload : false;
```

Comment:
Why the parameter is define to `null` if we want to not autoloaded?

Reply:
@mukeshpanchal27 , If we check the `add_option` function in the core, the $autoload option is by default null, hence I have kept it aligned, if it's true it will be true, if someone explicitly mentions it to be false it will be false, else if it stays null it will be considered false.

This is for backward compatibility.